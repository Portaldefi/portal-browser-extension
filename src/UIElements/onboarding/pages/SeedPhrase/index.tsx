import React, { useCallback } from 'react';
import { useNavigate } from 'react-router-dom';
import { Header, Button, Grid, Segment, Form } from 'semantic-ui-react';

import { useAppSelector } from '../../hooks';

export default () => {
  const navigate = useNavigate();

  const phrase = useAppSelector(state => state.phrase);
  console.log(phrase);

  const handleContinue = useCallback(() => {
    navigate('/seed-phrase-restore');
  }, []);
  const phraseRenderer = useCallback((phrases: Array<string>) => {
    let dispPhrases: Array<Array<string>>;

    dispPhrases = phrases.reduce((rows: Array<Array<string>>, phrase: string, index: number) => {
      if (index % 3 === 0) {
        rows.push([]);
      }

      rows.at(-1)?.push(phrase);

      return rows;
    }, []);

    return (
      <Form className='form-container'>
        {dispPhrases.map((rowPhrases, rowIdx) => (
          <Form.Group inline widths={3} key={rowIdx}>
            {rowPhrases.map((phrase, colIdx) => (
              <Form.Input label={`${rowIdx * rowPhrases.length + colIdx + 1}.`} width={16} readOnly value={phrase} key={colIdx} />
            ))}
          </Form.Group>
        ))}
      </Form>
    )
  }, []);

  return (
    <Segment className='board'>
      <Grid>
        <Grid.Row centered>
          <Header size='medium' className='heading'>New Seed Phrase</Header>
        </Grid.Row>
        <Grid.Row centered>
          {phraseRenderer(phrase.SRF_List)}
        </Grid.Row>
        <Grid.Row>
          <Header as='p' size='small' color='grey' className='description extra-former-blank extra-latter-blank'>
            WARNING: Improper backup of the seed phrase will result in loss of funds!
          </Header>
        </Grid.Row>
        <Grid.Row centered columns={1}>
          <Grid.Column>
            <Button className='primary-button' onClick={handleContinue}>Continue</Button>
          </Grid.Column>
        </Grid.Row>
      </Grid>
    </Segment>
  )
}