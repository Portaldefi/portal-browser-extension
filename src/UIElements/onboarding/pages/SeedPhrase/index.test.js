import SeedPhrase from './index';
import { create, act } from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store'

describe('SeedPhraseComponent', () => {
  const initialState = {
    phrase: {
      SRF_Length: 0,
      SRF_List: [],
      password: ''
    },
    menu: {
      items: []
    }
  }
  const mockStore = configureStore()
  let store


  it('should work', async () => {
    let tree;
    act(() => {
      store = mockStore(initialState)
      tree = create(
        <Provider store={store}>
          <BrowserRouter>
            <SeedPhrase />
          </BrowserRouter>
        </Provider>
      );
    });

    expect(tree).toMatchSnapshot();
  });
  afterAll(() => jest.resetModules());
});