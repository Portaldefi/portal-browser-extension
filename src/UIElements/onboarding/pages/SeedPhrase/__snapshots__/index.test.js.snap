// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SeedPhraseComponent should work 1`] = `
<div
  className="ui segment board"
>
  <div
    className="ui grid"
  >
    <div
      className="centered row"
    >
      <div
        className="ui medium header heading"
      >
        New Seed Phrase
      </div>
    </div>
    <div
      className="centered row"
    >
      <form
        className="ui form form-container"
        onSubmit={[Function]}
      />
    </div>
    <div
      className="row"
    >
      <p
        className="ui grey small header description extra-former-blank extra-latter-blank"
      >
        WARNING: Improper backup of the seed phrase will result in loss of funds!
      </p>
    </div>
    <div
      className="centered one column row"
    >
      <div
        className="column"
      >
        <button
          className="ui button primary-button"
          onClick={[Function]}
        >
          Continue
        </button>
      </div>
    </div>
  </div>
</div>
`;
